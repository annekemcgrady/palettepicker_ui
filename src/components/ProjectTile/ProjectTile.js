import React from 'react';
import './ProjectTile.scss'

const ProjectTile = ({ name,id, palettes, deletePalette, deleteProject, setPalette}) => {
    
    const projectPalettes = palettes.map(palette => {
      
        return (
            <article key={Date.now()} onClick={()=>setPalette(palette)} className='project-palette'>
                <p>{palette.name}</p>
                <div className='project-palette-div' style={{backgroundColor: palette.color_one}}></div>
                <div className='project-palette-div' style={{backgroundColor: palette.color_two}}></div>
                <div className='project-palette-div' style={{backgroundColor: palette.color_three}}></div>
                <div className='project-palette-div' style={{backgroundColor: palette.color_four}}></div>
                <div className='project-palette-div' style={{backgroundColor: palette.color_five}}></div>
                <button className='project-palette-delete' onClick={()=> deletePalette(palette.id)}>Delete</button>
            </article>
        )
    })

    return (
        <article className='project-tile'>
            <h2>Project: {name}</h2>
            {projectPalettes}
            <button className='project-delete' onClick={()=> deleteProject(id)}>Delete Project</button>
        </article>
    )
}

export default ProjectTile;