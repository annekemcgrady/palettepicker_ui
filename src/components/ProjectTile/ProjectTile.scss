.project-tile {
  width: 90%;
  background-color:aliceblue;
  border: 1px solid black;
  border-radius: 5px;
  margin: 5px;
}

.project-palette {
  font-size: 1em;
  display: flex;
  width: 90%;
  border: 1px solid black;
}

.project-palette-div {
  width: 20px;
  height: 20px;
  border: 1px solid black;
}

.project-palette-delete {
  width: 40px;
  height: 20px;
}