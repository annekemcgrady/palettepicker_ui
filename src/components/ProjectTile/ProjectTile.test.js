import React from 'react';
import ProjectTile from './ProjectTile';
import { shallow } from 'enzyme';


describe('ProjectTile', () => {
  let wrapper;
  let instance;
  let props;

  it('should match the snapshot', ()=>{

  })


})