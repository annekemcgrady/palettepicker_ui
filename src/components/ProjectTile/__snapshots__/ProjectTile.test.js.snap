// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTile should match the snapshot 1`] = `undefined`;
